projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewService.java

```diff
@@ -81,55 +83,76 @@ public class ViewService {
                FILTER NOT EXISTS { ?subject fs:dateDeleted ?anyDateDeleted }
             }
             """, FS.NS));
+    public static final String USER_DOES_NOT_HAVE_PERMISSIONS_TO_READ_FACETS = "User does not have permissions to read facets";
 
     private final Config.Search searchConfig;
+    private final Config.Caches cachesConfig;
     private final ViewsConfig viewsConfig;
     private final Dataset ds;
     private final ViewStoreClientFactory viewStoreClientFactory;
     private final MetadataPermissions metadataPermissions;
-    private final LoadingCache<Boolean, List<FacetDTO>> facetsCache;
-    private final LoadingCache<Boolean, List<ViewDTO>> viewsCache;
+    private LoadingCache<Boolean, List<FacetDTO>> facetsCache;
+    private LoadingCache<Boolean, List<ViewDTO>> viewsCache;
 
-    public ViewService(Config.Search searchConfig,
+    public ViewService(Config config,
                        ViewsConfig viewsConfig,
                        Dataset ds,
                        ViewStoreClientFactory viewStoreClientFactory,
                        MetadataPermissions metadataPermissions) {
-        this.searchConfig = searchConfig;
+        this.searchConfig = config.search;
+        this.cachesConfig = config.caches;
         this.viewsConfig = viewsConfig;
         this.ds = ds;
         this.viewStoreClientFactory = viewStoreClientFactory;
         this.metadataPermissions = metadataPermissions;
-        this.facetsCache = buildFacetCache();
-        this.viewsCache = buildViewsCache();
+        if (cachesConfig.facets.enabled) {
+            this.facetsCache = buildCache(this::fetchFacets, cachesConfig.facets);
+        }
+        if (cachesConfig.views.enabled) {
+            this.viewsCache = buildCache(this::fetchViews, cachesConfig.views);
+        }
         refreshCaches();
```

Comment:
I am a bit confused about the order of operations in this init.
The caches are first build, then refresh on startup, right?
In my logs I see:

> 2024-01-24 11:57:02 INFO  Caches refreshing/warming up has been triggered
2024-01-24 11:57:02 INFO  List of facets has been cached, 1 facets in total
2024-01-24 11:57:02 INFO  List of views has been cached, 4 views in total
2024-01-24 11:57:02 INFO  Caches refreshing/warming up successfully finished

which is a mix of the two operations: first start refreshing -> then build -> then finish refreshing

Is this for sure correct?

Reply:
`this::fetchViews` does not call the method right away, it is just a reference, so the build method just keeps the ref. the fetch method is called on refresh

Reply:
please contact me if still not clear, resolving not to be blocked